addons/common/functions/fnc_statusEffect_sendEffects.sqf

```diff
@@ -21,31 +21,34 @@ TRACE_2("params",_object,_effectName);
 
 if (isNull _object) exitWith {};
 
-{
-    if ((_effectName == "") || {_effectName == _x}) then {
-        private _effectVarName = format [QGVAR(effect_%1), _x];
-        private _effectNumber = _object getVariable [_effectVarName, -1];
+if (_effectName == "") exitWith { // recurse though all possible effects
+    {
+        [_object, _x] call FUNC(statusEffect_sendEffects)
+    } forEach (keys GVAR(statusEffects)); 
+};
 
-        //We only do anything if the effect has been defined at some point in the game for this unit
-        TRACE_2("checking if event is nil",_x,_effectNumber);
-        if (_effectNumber != -1) then {
-            private _eventName = format [QGVAR(%1), _x];
-            switch (true) do {
-                case (GVAR(statusEffect_sendJIP) select _forEachIndex): {
-                    TRACE_2("Sending Global JIP Event",_object,_effectNumber);
-                    private _jipID = format [QGVAR(effect_%1_%2), _eventName, hashValue _object];
-                    [_eventName, [_object, _effectNumber], _jipID] call CBA_fnc_globalEventJIP;
-                    [_jipID, _object] call CBA_fnc_removeGlobalEventJIP;
-                };
-                case (GVAR(statusEffect_isGlobal) select _forEachIndex): {
-                    TRACE_2("Sending Global Event",_object,_effectNumber);
-                    [_eventName, [_object, _effectNumber]] call CBA_fnc_globalEvent;
-                };
-                default {
-                    TRACE_2("Sending Target Event",_object,_effectNumber);
-                    [_eventName, [_object, _effectNumber], _object] call CBA_fnc_targetEvent;
-                };
-            };
+
+private _effectVarName = format [QGVAR(effect_%1), _effectName];
+private _effectNumber = _object getVariable [_effectVarName, -1];
+
+//We only do anything if the effect has been defined at some point in the game for this unit
+TRACE_2("checking if event is nil",_effectName,_effectNumber);
+if (_effectNumber != -1) then {
+    (GVAR(statusEffects) get toLowerANSI _effectName) params ["_isGlobal", "_sendJIP", "_eventName"];
```

Comment:
Do we want to support non-ANSI effect names too or not?

Reply:
IMO non-ANSI effect names would be totally unreasonable.

Reply:
It doesn't matter if it's reasonable or not, it matters if it's possible. Given that `addType` forces all names to be `toLowerANSI`, it's impossible and therefore doesn't matter.